smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -799,7 +915,15 @@ private void readHeaders(
         // There can only be one payload binding.
         HttpBinding binding = payloadBindings.get(0);
         Shape target = context.getModel().expectShape(binding.getMember().getTarget());
-        if (hasStreamingComponent) {
+        if (binding.getMember().hasTrait(EventStreamTrait.class)) {
+            // If payload is a event stream, return it after calling event stream deser function.
+            writer.write("const data: any = ").indent();
```

Comment:
This can be combined with the `openBlock` call in the function call that follows, removing the need to do indent/dedent.